src/site/markdown/development/build-cmake.md

```diff
@@ -22,28 +22,82 @@ Building with CMake {#build-cmake}
 -->
 [TOC]
 
-## Quick start:
+# Supported options
 
-  Building and testing Log4cxx on a Unix platform with packaged APR and APR-Util.
-
-  Make sure cmake 3.13+, g++ and make are available, install or
-  build apr 1.x, apr-util 1.x, gzip and zip.
+| Option                 | Usage |
+|------------------------|-------|
+| LOG4CXX_WCHAR_T=no   | Enable wchar_t API methods, choice of yes (default), no.                                    |
+| LOG4CXX_UNICHAR=yes  | Enable UniChar API methods, choice of yes, no (default).                                    |
+| LOG4CXX_CFSTRING=yes | Enable CFString API methods, requires Mac OS/X CoreFoundation, choice of yes, no (default). |
+| BUILD_TESTING=off    | Do not build tests.  Tests are built by default                                             |
+| BUILD_SHARED_LIBS=off| Build Log4cxx as a static library. A dynamically linked Log4cxx library is built by default. Any compilation unit that includes a Log4cxx header must define LOG4CXX_STATIC.             |
+| APU_STATIC=yes       | Link to the APR-Util static library. By default, the Log4cxx shared library is linked to the APR-Util shared library. If BUILD_SHARED_LIBS=off, the static APR-Util library is used.     |
+| APR_STATIC=yes       | Link to the APR static library. By default, the Log4cxx shared library is linked to the APR shared library. If BUILD_SHARED_LIBS=off, the static APR library is always used.        |
+| LOG4CXX_TEST_PROGRAM_PATH=path | An extra path to prepend to the PATH for test programs.  Log4cxx requires zip, sed, and grep on the PATH in order for the tests to work properly.                          |
+| PREFER_BOOST=on      | Prefer the Boost version of dependent libraries over standard library |
+| LOG4CXX_QT_SUPPORT=ON | Enable QString API and log4cxx::qt namespace methods, requires QtCore, choice of ON, OFF (default).                   |
+| LOG4CXX_EVENTS_AT_EXIT=ON | Prevent static data cleanup to allow event logging during application exit. |
+| LOG4CXX_FORMAT_NAMESPACE=std | Make the `LOG4CXX_[level]_FMT` macros use [std::format](https://en.cppreference.com/w/cpp/utility/format/format) by default.  |
+
+# Examples
+
+## Unix type systems
+
+Make sure cmake (3.13+), g++, gzip, zip and make are available.
+On a Debian system these may be installed by:
+~~~
+$ sudo apt-get install build-essential cmake gzip zip
+~~~
 
-Linux example:
+1. Using the distro provided APR libraries
+and the cmake cursors UI to configure Log4cxx options.
 ~~~
-$ apt-get install build-essential libapr1-dev libaprutil1-dev gzip zip
-$ cd apache-Log4cxx-x.x.x
+$ apt-get install libapr1-dev libaprutil1-dev
+$ wget https://dlcdn.apache.org/logging/log4cxx/1.3.0/apache-log4cxx-1.3.0.tar.gz
```

Comment:
Do we do a CMake replacement here to always point at the latest version?

Reply:
> Do we do a CMake replacement here to always point at the latest version?

Not necessary I think so I prefer to no make it even more complicated. CMakeList.txt says `cmake_minimum_required(VERSION 3.13)`. Even Ubuntu 20.04 distro includes CMake 3.16.